tests/integration/lifecycle/test_overlay.py

```diff
@@ -35,7 +35,13 @@ def fake_call(mocker):
 def mock_overlay_support_prerequisites(mocker):
     mocker.patch.object(sys, "platform", "linux")
     mocker.patch("os.geteuid", return_value=0)
-    mocker.patch("craft_parts.overlays.OverlayManager.refresh_packages_list")
+    mock_refresh = mocker.patch(
+        "craft_parts.overlays.OverlayManager.refresh_packages_list"
+    )
+    yield
+    # Make sure that refresh_packages_list() was *not* called, as it's an expensive call that
+    # overlays without packages do not need.
+    assert not mock_refresh.called
```

Comment:
this one is just to be double-extra-sure, heh

Reply:
heh, negative assertions can become eternal

Reply:
I think I know what you mean, but can you elaborate? Should I remove this assertions?

Reply:
sorry, it was a side comment :-)